deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkUri.java

```diff
@@ -1215,7 +1217,22 @@ private static boolean decodeIpv4Suffix(
      */
     private static String domainToAscii(String input) {
       try {
-        String result = IDN.toASCII(input).toLowerCase(Locale.US);
+        // When this is a template URI it might get to long with all the placeholder definitions
+        // to pass verification.
+        // This will replace the placeholders with other shorter placeholders for URI validation.
+        Set<String> placeholders = getPlaceHolders(input);
+        String mappedInput = input;
+        Map<String, String> placeholderMap = new HashMap(placeholders.size());
+        int i = 0;
+        for (String placeholder : placeholders) {
```

Comment:
would be really nice to have this file in kotlin (not in this PR of course, but one day :P )

Reply:
Well the reason it is not is because it is literally a copy of another file. That is why I never ported it.